### PR TITLE
tests: path quoting and boilerplate

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
All tests now run when there are spaces in `$BASH_IT`, and boilerplate removed from the top of most test files. 

## Motivation and Context
As I'm learning BATS, I'm cleaning up duplicate code and hopefully making the tests faster and more reliable. There's more to do! This is just my first pass.

## How Has This Been Tested?
I run the tests manually myself, and alsö through GitHub Actions. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Comment by gaelicWizard** [comment]
:grumble grumble: the OSX.11 test runner is so slow that it fails the `plugin/cmd-returned-notify` test!!

File: test/plugins/ruby.plugin.bats [hunk 1]
```diff
@@ -18,17 +18,21 @@ function local_teardown {
 }
 
 @test "plugins ruby: remove_gem is defined" {
+  load ../../plugins/available/ruby.plugin
```
Comment (NoahGorny) [inline test/plugins/ruby.plugin.bats:21]: isnt this duplicated? why do we need to load again?

Comment (gaelicWizard) [reply test/plugins/ruby.plugin.bats:19]: Fixed!

**Review by NoahGorny — APPROVED** [review]
We could add `test/run` to `clean_files.txt` as well

**Review by NoahGorny — APPROVED** [review]
Horray!
Great work @gaelicWizard

**Pull request merged**